chartfx-dataset/src/main/java/de/gsi/dataset/DefaultNumberFormatter.java

```diff
@@ -0,0 +1,256 @@
+package de.gsi.dataset;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Locale;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * DefaultNumberFormatter implementing the Formatter&lt;T&gt; interface.
+ *
+ * The number representation can be set via #setFormatMode
+ * <ul>
+ * <li> FIXED_WIDTH_ONLY: using decimal representations only. The string width is set via #setNumberOfCharacters(int)</li>
+ * <li> FIXED_WIDTH_EXP: using exponential representation only. The string width is set via #setNumberOfCharacters(int)</li>
+ * <li> FIXED_WIDTH_AND_EXP: using decimal or exponential representations, with preference order: shorter representation, more significant digits.
+ *         The string width is set via #setNumberOfCharacters(int)
+ *         N.B. This mode is only useful for #getNumberOfCharacters width greater than 6 (due to 'E[+,-]0' overhead)
+ * </li>
+ * <li> OPTIMAL_WIDTH (default): using the shorter of decimal vs. exponential representation
+ *         Number of significant digits is set via #setFixedPrecision(int)
+ * </li>
+ * <li> METRIC_PREFIX: using standard SI/metric-unit prefixes.
+ *         Number of significant digits is set via #setFixedPrecision(int)
+ *         The default precision is 3, i.e. 1234.5678 -&gt; "1.234k"
+ * </li>
+ * <li> BYTE_PREFIX: using standard SI/metric-unit prefixes with base '1024'.</li>
+ * <li> JDK: JDK default using the 'toString()' method</li>
+ * </ul>
+ *
+ * @author rstein
+ * @see de.gsi.dataset.Formatter
+ */
+public class DefaultNumberFormatter implements Formatter<Number> {
+    protected static final int NO_PREFIX_OFFSET = 8;
+    protected static final String SI_PREFIX = "yzafpnµm kMGTPEZY";
+    protected static final String SI_PREFIX_TEST = "yzafpnuµmkKMGTPEZY"; // N.B. doubling of micro and kilo representation for parsing
+    protected static final int[] SI_PREFIX_EXP = { -24, -21, -18, -15, -12, -9, -6, -6, -3, 3, 3, 6, 9, 12, 15, 18, 21, 24 };
+    protected DecimalFormat decimalFormat = new DecimalFormat("#.#", DecimalFormatSymbols.getInstance(Locale.UK));
+    protected DecimalFormat decimalFormatMaxPrecision = new DecimalFormat("#.#", DecimalFormatSymbols.getInstance(Locale.UK));
+    protected String fixedLengthFormat;
+    protected String fixPrecisionFormat;
+    protected String fixPrecisionFormatZero;
+    private int numberOfCharacters = 6; // N.B. exp-form only useful for width>=6
```

Comment:
This is more qualitative and would (theoretically) work with less (i.e. '>= 4'). IMHO this isn't a very common/useful representation for most cases since, for example, '1' would end up as '1E+0'. Representations like '1.0E+0', '1.0E+9', or ''1.0E-9' would look more natural/common... but this is more a matter of taste and can be changed by the inclined user.

Adjusted the documentation accordingly.